server/controllers/finance/cashflow.js

```diff
@@ -0,0 +1,476 @@
+
+	return db.exec(requette, [params.account_id, params.dateFrom, params.dateTo, params.account_id, params.dateFrom, params.dateTo]);
+}
+
+/**
+ * @function groupingIncomeExpenseByPeriod
+ * @description This function help to group incomes or expenses by period
+ */
+function groupingIncomeExpenseByPeriod(periodicFlows) {
+	var grouping = [];
+	periodicFlows.forEach(function (pf) {
+		var incomes, expenses;
+		incomes = pf.flows.filter(function (posting) {
+			return posting.debit_equiv > 0;
+		});
+		expenses = pf.flows.filter(function (posting) {
+			return posting.credit_equiv > 0;
+		});
+		grouping.push({ period: pf.period, incomes : incomes, expenses : expenses });
+	});
+	return grouping;
+}
+
+/**
+ * @function groupingByMonth
+ * @param {array} periods An array which contains all periods for the fiscal year
+ * @param {array} flows The result of `queryIncomeExpense` i.e. all incomes and expense
+ * @description This function help to group incomes or expenses by month
+ */
+function groupingByMonth(periods, flows) {
+	var grouping = [];
+	periods.forEach(function (p) {
+		var data = [];
+		flows.forEach(function (f) {
+			var trans_date = new Date(f.trans_date);
+			var start_date = new Date(p.start_date);
+			var end_date = new Date(p.end_date);
+			if (trans_date <= end_date && trans_date >= start_date) {
+				data.push(f);
+			}
+		});
+		grouping.push({ period: p, flows : data });
+	});
+	return grouping;
+}
+
+/**
+ * @function closingBalance
+ * @param {number} accountId An account for which we search to know the balance
+ * @param {date} periodStart The first period start of a given fiscal year (current fiscal year)
+ * @desc This function help us to get the balance at cloture for a set of accounts
+ */
+function closingBalance(accountId, periodStart) {
+	var query =
+    'SELECT SUM(`debit_equiv` - `credit_equiv`) as balance, `account_id` '+
+    'FROM ' +
+    '(' +
+	    '(' +
+	    	'SELECT `debit_equiv`, `credit_equiv`, `account_id`, `currency_id` ' +
+	    	'FROM `posting_journal` ' +
+	    	'WHERE `account_id` IN (?) AND `fiscal_year_id` = ? ' +
+	    ') UNION ALL (' +
+	    	'SELECT `debit_equiv`, `credit_equiv`, `account_id`, `currency_id` ' +
+	    	'FROM `general_ledger` ' +
+	    	'WHERE `account_id` IN (?) AND `fiscal_year_id` = ? ' +
+	    ')' +
+	  ') as `t` ';
+
+    return getFiscalYear(periodStart)
+    .then(function (rows) {
+    	var fy = rows[0];
+    	return db.exec(query, [accountId, fy.previous_fiscal_year_id, accountId, fy.previous_fiscal_year_id]);
+    });
+}
+
+/**
+ * @function getFiscalYear
+ * @param {object} date The date in which we want to get the fiscal year
+ * @description
+ * This function is responsible of returning a correct fiscal year
+ * according a date given
+ */
+function getFiscalYear(date) {
+	var query =
+		'SELECT fy.id, fy.previous_fiscal_year_id FROM fiscal_year fy ' +
+		'JOIN period p ON p.fiscal_year_id = fy.id ' +
+		'WHERE ? BETWEEN p.start_date AND p.end_date';
```

Comment:
This JOIN works, but another way to do it without the JOIN would might be:
```sql
SELECT * FROM fiscal_year WHERE ? BETWEEN fy.start_date AND ADDDATE(fy.start_date, INTERVAL fy.number_of_months MONTH);
```
Note: I haven't tested this.  But thought it might be cool.